### PR TITLE
Validate the ResizePolicy field in Container

### LINKED ISSUE #116854: [InPlacePodVerticalScaling] ResizePolicy field is not being validated
### What happened?

https://github.com/kubernetes/kubernetes/pull/102884#discussion_r1144672887

The newly added field `ResizePolicy` field is not being validated.

@vinaykul @liggitt 

### What did you expect to happen?

`ResizePolicy` field should be validated.

### How can we reproduce it (as minimally and precisely as possible)?

We can create a pod with invalid `ResizePolicy` to reproduce this.

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by gjkim42**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

This validates the ResizePolicy field in Container

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/116854

#### Does this PR introduce a user-facing change?

```release-note
none
```

As the feature is not released yet.

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/cc @liggitt @vinaykul

### CODE REVIEW

**Comment by gjkim42** [comment]
updating tests...

**Comment by gjkim42** [comment]
@liggitt could you PTAL?

File: pkg/api/pod/util.go [hunk 1]
```diff
@@ -396,6 +432,7 @@ func GetValidationOptionsFromPodSpecAndMeta(podSpec, oldPodSpec *api.PodSpec, po
 		AllowInvalidLabelValueInSelector:                  false,
 		AllowInvalidTopologySpreadConstraintLabelSelector: false,
 		AllowMutableNodeSelectorAndNodeAffinity:           utilfeature.DefaultFeatureGate.Enabled(features.PodSchedulingReadiness),
+		AllowResizePolicy:                                 utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) || !hasNewOrUpdatedResizePolicy(podSpec, oldPodSpec),
```
Comment (liggitt) [inline pkg/api/pod/util.go:435]: drop changes to this file... we don't need this in validation options, we clear the field before reaching validation if the feature is disabled

File: pkg/apis/core/validation/validation_test.go [hunk 2]
```diff
@@ -7591,192 +7591,214 @@ func TestValidateContainers(t *testing.T) {
...
-					Exec: &core.ExecAction{Command: []string{"ls", "-l"}},
-				},
-			},
-			ImagePullPolicy:          "IfNotPresent",
-			TerminationMessagePolicy: "File",
-		},
+	successCases := []struct {
+		title, line string
+		containers  []core.Container
+		opts        PodValidationOptions
```
Comment (liggitt) [inline pkg/apis/core/validation/validation_test.go:7597]: can drop this change, we don't need a new validation option for this... if the field is set, validate it

**Comment by gjkim42** [comment]
I'll close this in favor of https://github.com/kubernetes/kubernetes/pull/116857

**Pull request closed without merging**